src/lib.rs

```diff
@@ -242,6 +242,35 @@ impl<T> Sender<T> {
         }
     }
 
+    /// Sends a message into this channel using the blocking strategy.
+    ///
+    /// If the channel is full, this method will block until there is room.
+    /// If the channel is closed, this method returns an error.
+    ///
+    /// # Blocking
+    ///
+    /// Rather than using asynchronous waiting, like the [`send`] method,
+    /// this method will block the current thread until the message is sent.
+    ///
+    /// This method should not be used in an asynchronous context. It is intended
+    /// to be used such that a channel can be used in both asynchronous and synchronous contexts.
+    /// Calling this method in an `async` block may result in deadlocks.
```

Comment:
```suggestion
    /// Calling this method in an asynchronous context may result in deadlocks.
```

I think the "asynchronous context" you are already using in the above line is correct here, since using it in manually written futures would cause the same problem.